common/configuration/asm_test.go

```diff
@@ -0,0 +1,128 @@
+package configuration
+
+import (
+	"context"
+	"fmt"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/secretsmanager"
+	"sort"
+	"testing"
+
+	"github.com/TBD54566975/ftl/internal/log"
+
+	"github.com/alecthomas/assert/v2"
+	. "github.com/alecthomas/types/optional"
+)
+
+func localstack(ctx context.Context, t *testing.T) ASM {
+	t.Helper()
+	cc := aws.NewCredentialsCache(credentials.NewStaticCredentialsProvider("test", "test", ""))
+	cfg, err := config.LoadDefaultConfig(ctx, config.WithCredentialsProvider(cc), config.WithRegion("us-west-2"))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	sm := secretsmanager.NewFromConfig(cfg, func(o *secretsmanager.Options) {
+		o.BaseEndpoint = aws.String("http://localhost:4566")
+	})
+	asm := ASM{client: *sm}
+	return asm
+}
+
+func TestASMWorkflow(t *testing.T) {
+	ctx := log.ContextWithNewDefaultLogger(context.Background())
+	asm := localstack(ctx, t)
```

Comment:
I think it would be preferable to test this resolver+provider with the `Manager` directly, just to make sure everything is working end-to-end.